core/services/workflows/engine.go

```diff
@@ -747,6 +753,7 @@ func (e *Engine) worker(ctx context.Context) {
 			if err != nil {
 				e.logger.With(platform.KeyWorkflowExecutionID, executionID).Errorf("failed to start execution: %v", err)
 				logCustMsg(ctx, cma, fmt.Sprintf("failed to start execution: %s", err), e.logger)
```

Comment:
Unrelated to this PR but I wonder if logCustMsg() could include a logger call so we could only have a single line for both? cc @patrickhuie19

Reply:
^ I originally had that in the initial implementation, but ultimately decided against that to decouple the logging functionality and custom messages. It got messy b/c either the utility is streamlined and decides a format of the message for the consumer, or it is flexible and accepts a format string. The former case was undesirable, and the latter case relies on variadic args. Go doesn't support more than one variadic param in a func, so only one format could be applied to both the log and the custom message.